app/uk/gov/hmrc/pillar2externalteststub/controllers/SubscriptionController.scala

```diff
@@ -34,25 +32,9 @@ class SubscriptionController @Inject() (
 ) extends BackendController(cc)
     with Logging {
 
-  def retrieveSubscription(plrReference: String): Action[AnyContent] =
-    (Action andThen authFilter).async { implicit request =>
-      logger.info(s"Retrieving subscription for PLR reference: $plrReference")
-
-      plrReference match {
-        case "XEPLR0123456404" =>
-          Future.successful(NotFound(Json.toJson(NotFoundSubscription.response)))
-        case "XEPLR0123456500" =>
-          Future.successful(InternalServerError(Json.toJson(ServerError500.response)))
-        case "XEPLR0123456503" =>
-          Future.successful(ServiceUnavailable(Json.toJson(ServiceUnavailable503.response)))
-        case "XEPLR5555555555" =>
-          Future.successful(Ok(Json.toJson(SubscriptionSuccessResponse.successfulDomesticOnlyResponse(plrReference))))
-        case "XEPLR1234567890" =>
-          Future.successful(Ok(Json.toJson(SubscriptionSuccessResponse.successfulNonDomesticResponse(plrReference))))
-        case "XEPLR0987654321" =>
-          Future.successful(Ok(Json.toJson(NilReturnSuccess.successfulResponse)))
```

Comment:
Also documented in the README, I'm unsure why a NilReturnSuccess is being returned from this controller and not SubmitUKTRController, isn't the Nil Return a subtype of UKTR?

<img width="315" alt="Screenshot 2024-12-15 at 19 43 31" src="https://github.com/user-attachments/assets/c18f14dd-f694-4bea-b03e-f5f1ee32d308" />